.github/workflows/documentation_workflow.yml

```diff
@@ -0,0 +1,63 @@
+name: Documentation
+on: [push, pull_request]
+
+jobs:
+  MakeDocs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup R
+        uses: r-lib/actions/setup-r@master
+        with:
+          r-version: '4.1.0'
+      - name: Create Documentation directory
+        run: |
+          echo "Creating 'Documentation'"
+          mkdir -p Documentation
+      - name: Install knitr dependency
+        run: |
+          echo "Installing R package dependencies. Scripts might have additional dependencies installed."
+          Rscript -e 'if (!require(knitr)) install.packages("knitr")'
+      - name: Documentation compilation
+        run: |
+          Rscript -e '
+          cat("R process started.\n")
+          cat("Change working directory to documentation directory\n")
+          setwd("Documentation")
+
+          cat("Creating the directory list\n")
+          dirlist <- c("../Association-Algorithms",
+                       "../Classification-Algorithms",
+                       "../Clustering-Algorithms",
+                       "../Data-Manipulation",
+                       "../Data-Mining",
+                       "../Data-Preprocessing",
+                       "../Machine-Learning",
+                       "../Mathematics",
+                       "../Regression-Algorithms",
+                       "../Sorting-Algorithms")
+          print(dirlist)
+
+          cat("Getting a list of R scripts from the algorithm directories.\n")
+          scriptlist <- lapply(dirlist, list.files, ".R", full.names = TRUE)
+          cat("Removing from the list empty directories.\n")
+          scriptlist <- scriptlist[!sapply(scriptlist, identical, character(0))]
+          print(unlist(scriptlist))
+
+          cat("Compiling documentation from scripts.\n")
+          invisible(lapply(unlist(scriptlist), function(x) tryCatch(knitr::spin(x),
+                                                                    error = function(e) message("Error compiling: ", x))))
+
+          cat("Removing superfluous markdown files.\n")
+          invisible(file.remove(list.files(pattern = ".md")))
```

Comment:
```suggestion
```
I suggest leaving .md files because they can be reviewed in PR much easier than html.